CMakeLists.txt

```diff
@@ -82,27 +101,28 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "^arm" OR
     message(FATAL_ERROR
             "PNG_ARM_NEON must be one of [${PNG_ARM_NEON_POSSIBLE_VALUES}]")
   elseif(NOT ${PNG_ARM_NEON} STREQUAL "off")
-    set(libpng_arm_sources
```

Comment:
Why has it to be moved? For minimal patch purposes, it should be kept here.

Reply:
The hardware specific code should always be included, even if it is not for the target hardware. The code will preprocess itself away if it does not match the current target hardware. For iOS, multiple hardware is targeted so `CMAKE_SYSTEM_PROCESSOR` is not set and **no** optimized source was included.

Reply:
> For iOS, multiple hardware is targeted so CMAKE_SYSTEM_PROCESSOR is not set and no optimized source was included

For combined iOS build we will enter this `if` too, see `OR CMAKE_SYSTEM_PROCESSOR STREQUAL ""` condition.